test/net-test.js

```diff
@@ -0,0 +1,2218 @@
+        map.set(packets.types.PONG, 'handlePong');
+        map.set(packets.types.GETADDR, false);
+        map.set(packets.types.ADDR, false);
+        map.set(packets.types.INV, false);
+        map.set(packets.types.GETDATA, false);
+        map.set(packets.types.NOTFOUND, false);
+        map.set(packets.types.GETBLOCKS, false);
+        map.set(packets.types.GETHEADERS, false);
+        map.set(packets.types.HEADERS, false);
+        map.set(packets.types.SENDHEADERS, 'handleSendHeaders');
+        map.set(packets.types.BLOCK, false);
+        map.set(packets.types.TX, false);
+        map.set(packets.types.REJECT, false);
+        map.set(packets.types.MEMPOOL, false);
+        map.set(packets.types.FILTERLOAD, 'handleFilterLoad');
+        map.set(packets.types.FILTERADD, 'handleFilterAdd');
+        map.set(packets.types.FILTERCLEAR, 'handleFilterClear');
+        map.set(packets.types.MERKLEBLOCK, false);
+        map.set(packets.types.FEEFILTER, 'handleFeeFilter');
+        map.set(packets.types.SENDCMPCT, 'handleSendCmpct');
+        map.set(packets.types.CMPCTBLOCK, false);
+        map.set(packets.types.GETBLOCKTXN, false);
+        map.set(packets.types.BLOCKTXN, false);
+        map.set(packets.types.UNKNOWN, false);
+        map.set(packets.types.INTERNAL, false);
+        map.set(packets.types.DATA, false);
+
+        const wrap = (type, handler) => {
+          const peer = Peer.fromOptions({});
+          const result = {count: 0, peer};
+
+          for (const fn of map.values()) {
+            if (fn) {
+              peer[fn] = (packet) => {
+                assert.equal(fn, handler);
+                assert(packet);
+                assert.equal(packet.type, type);
+                result.count += 1;
+              };
+            }
+          }
+
+          return result;
+        };
+
+        for (const [type, handler] of map) {
+          const stub = wrap(type, handler);
+          const packet = {type};
+
+          await stub.peer.handlePacket(packet);
+          if (handler)
+            assert.equal(stub.count, 1);
+          else
+            assert.equal(stub.count, 0);
+        }
+      });
+    });
+
+    describe('handleVersion', function() {
+      it('will error if already sent version', async () => {
+        const peer = Peer.fromOptions({});
+        peer.version = 1000;
+        const pkt = new packets.VersionPacket();
+        let err = null;
+
+        try {
+          await peer.handleVersion(pkt);
+        } catch (e) {
+          err = e;
+        }
+        assert(err);
+      });
+
+      it('will not connect to self', async () => {
+        const peer = Peer.fromOptions({});
+        peer.options.hasNonce = () => true;
+
+        const pkt = new packets.VersionPacket();
+        let err = null;
+
+        try {
+          await peer.handleVersion(pkt);
+        } catch (e) {
+          err = e;
+        }
+        assert(err);
+        assert.equal(err.message, 'We connected to ourself. Oops.');
```

Comment:
Since this is network specific, do you want to add something like `assert(peer.network === 'main')` or `assert(!peer.network.selfConnect)` or something to prevent regression?